content/articles/data-preprocessing-python/index.md

```diff
@@ -0,0 +1,248 @@
+In this article, we will learn all the steps that are involved in the data preprocessing phase. 
+
+### Prerequisites
+
+To follow this tutorial comfortably, we need to meet the following conditions.
+1. Prior knowledge in python programming.
+2. Connected to the internet.
+2. Access to Google Colab [Here](https://colab.research.google.com/notebooks/intro.ipynb).
+
+### Mounting our drive to Google Colab
+
+In this article, we shall carry out our data preprocessing activities on Google Colad. Therefore, we need to ensure our Google Drive is accessible by the Google Colab. To ensure this, first, let's download our data to our computer from [here](https://github.com/Madrinmarison/data).
+
+Since we've successfully downloaded our data, let's now upload it to google drive through [google.drive.com](https://drive.google.com/drive/u/0/my-drive).
+
+Now our data is uploaded to Google Drive.
+
+To be able to use this data, we need to give Google Colab access to the drive. To enhance this, in Google Colab, let's type and run the code below.
+
+```python
+from google.colab import drive
+drive.mount("/content/drive/")
+```
+Upon executing our code, it will lead us to a Google Authentication stage similar to the figure below.
+
+![Google_Authentication](/engineering-education/data-preprocessing-python/URL.png)
+
+We click on the URL link on your Google Colab interface and proceed to allow permissions until we reach the verification code stage, i.e.
+
+![Verification_code](/engineering-education/data-preprocessing-python/Cpo.png)
+
+We then copy and paste this code in the authorization code box on Colab interface and click **Ctrl + Enter**. This shows that our drive is mounted.
+
+Now that our drive it's successfully mounted in Google Colab, we proceed and import the libraries we shall require to work with along our journey.
+To import these libraries, let's type and run the code below.
+
+
+### Step 1: Importing the Libraries
+
+```Python
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+```
+Next, we proceed and import the dataset.
+
+### Step 2: Import the dataset
+
+Let's type and run the following code.
+
+```python
+Dataset = pd.read_csv("/content/drive/MyDrive/Dataset.csv")
+
+# importing an array of features
+x = Dataset.iloc[:, :-1].values 
+# importing an array of dependent variable
+y = Dataset.iloc[:, -1].values
+```
+
+Above, we specified two variables, x for the features and y for the dependent variable. The features set, as declared in the code `Dataset.iloc[:, :-1]` will consist of all rows and columns of our dataset except the last column. Similarly, the dependent variable y will consist of all rows but only the last column as declared in the code `Dataset.iloc[:, -1].values`
+
+Let's have a look at our data by executing the code.
+
+```Python
+print(x) # returns an array of features
+```
+
+![Features_Array](/engineering-education/data-preprocessing-python/Feature.array.png)
```

Comment:
The terminal output can be shown as a formatted block instead of the screenshot.

For example, 

```bash
> x
[[1,2],[3,4]]
```

Just copy the terminal output and paste it in the formatted block.